### DISCUSSION TITLE
creating tables relationships

### DISCUSSION THREAD

**Discussion by ounced** [AntlerVC/firetable#1435] [Support / Q&A] [answered]
Hi there, i'm building a recipe library app and i've created a few tables in Rowy. I've set up algolia to enable the connect table column type. The goal is for me to link a tag in "recipeTags" table to "recipeList" table using recipeID. 

recipeList
<img width="906" alt="Screenshot 2023-10-16 at 4 34 12 PM" src="https://github.com/rowyio/rowy/assets/147575554/5513b4ee-45f7-4c42-82c0-e6fd551442ee">

recipeTag
<img width="1059" alt="Screenshot 2023-10-16 at 4 34 21 PM" src="https://github.com/rowyio/rowy/assets/147575554/dd81c860-424f-4823-9ac0-d61bd2962bc0">

My question is:

1. if algolia doesn't work, what method can i link up the table together?
2. how can i do this for multiple recipes in one go or make a bulk edit? (in airtable, i can link records using an extension or convert the field to a linked record and all records will be linked)
3. if i make the relationship in firestore, will i be able to see them in rowy?
4. if i create the relationship in airtable, can the relationship be imported in rowy? 

Thank you!

**Comment by Chadha93** [comment, marked as answer]
Hi @ounced. Answering your questions:

1. If Algolia doesn't work, you can still create a relationship between the "recipeTags" table and the "recipeList" table using the recipeID. One way to do this is by adding a column in the "recipeTags" table that stores the recipeID from the "recipeList" table. You can then use this column to link the two tables together.

Here's an example of how you can create the relationship using a JavaScript GCP Cloud Function:

```javascript
const admin = require('firebase-admin');

exports.createRelationship = async (req, res) => {
  const recipeID = req.body.recipeID; // RecipeID to link the tables
  const tagID = req.body.tagID; // ID of the tag in the "recipeTags" table

  try {
    // Get the recipe document from the "recipeList" table
    const recipeDoc = admin.firestore().collection('recipeList').doc(recipeID);
    const recipeSnapshot = await recipeDoc.get();
    const recipeData = recipeSnapshot.data();

    // Add the tagID to the recipe document
    await recipeDoc.update({
      tagID: tagID
    });

    res.status(200).send('Relationship created successfully');
  } catch (error) {
    console.error('Error creating relationship:', error);
    res.status(500).send('Error creating relationship');
  }
};
```

2. To create the relationship for multiple recipes in one go or make a bulk edit, you can modify the above code to accept an array of recipeIDs and tagIDs. You can then loop through the array and create the relationship for each recipe.

```javascript
exports.createRelationshipBulk = async (req, res) => {
  const relationships = req.body.relationships; // Array of recipeID-tagID pairs

  try {
    for (const { recipeID, tagID } of relationships) {
      const recipeDoc = admin.firestore().collection('recipeList').doc(recipeID);
      const recipeSnapshot = await recipeDoc.get();
      const recipeData = recipeSnapshot.data();

      await recipeDoc.update({
        tagID: tagID
      });
    }

    res.status(200).send('Bulk relationships created successfully');
  } catch (error) {
    console.error('Error creating bulk relationships:', error);
    res.status(500).send('Error creating bulk relationships');
  }
};
```

3. If you make the relationship in Firestore, you will be able to see them in Rowy. Rowy syncs with Firestore and allows you to view and edit the data in your Firestore collections and documents.

4. If you create the relationship in Airtable, you won't be able to directly import the relationship into Rowy. However, you can export the data from Airtable and import it into Firestore using a data migration tool or by writing a script to transform and import the data. Once the data is in Firestore, you can view and manage the relationships in Rowy.

I hope the above solutions/answers will be helpful. Feel free to reach out if you have more questions or feedback. You can also join one of our upcoming [office hours](https://buildship.com/events) and [Discord](https://buildship.com/discord) to interact with the community.

--
Cheers,
Gaurav Chadha | Technical Support Engineer | [Rowy](https://rowy.app/) + [BuildShip](https://buildship.com/)
backend for everyone

**Comment by ounced** [comment]
Hey Gaurav, thanks for this. let me try this out. i'm not very technical person but is learning. will need some time to implement this solution.
